src/core/caniuse.js

```diff
@@ -0,0 +1,192 @@
+/*
+Module: "core/caniuse"
+Adds a caniuse support table for a "feature" #1238
+https://github.com/w3c/respec/wiki/caniuse
+*/
+
+import { semverCompare } from "core/utils";
+import { pub } from "core/pubsubhub";
+import "deps/hyperhtml";
+import { createResourceHint, fetchAndCache } from "core/utils";
+import caniuseCss from "deps/text!core/css/caniuse.css";
+
+export const name = "core/caniuse";
+
+const GH_USER_CONTENT_URL =
+  "https://raw.githubusercontent.com/Fyrd/caniuse/master/features-json/";
+
+const BROWSERS = new Map([
+  // browser name dictionary
+  ["chrome", "Chrome"],
+  ["firefox", "Firefox"],
+  ["ie", "IE"],
+  ["edge", "Edge"],
+  ["android", "Android"],
+  ["safari", "Safari"],
+  ["opera", "Opera"],
+  ["bb", "Blackberry"],
+  ["and_uc", "UC (Android)"],
+  ["and_ff", "Firefox (Android)"],
+]);
+
+// Keys from https://github.com/Fyrd/caniuse/blob/master/CONTRIBUTING.md
+const supportTitles = new Map([
+  ["y", "Supported."],
+  ["a", "Almost supported (aka Partial support)."],
+  ["n", "No support, or disabled by default."],
+  ["p", "No support, but has Polyfill."],
+  ["u", "Support unknown."],
+  ["x", "Requires prefix to work."],
+  ["d", "Disabled by default (needs to enabled)."],
+]);
+
+export function run(conf) {
+  if (!conf.caniuse) {
+    return; // nothing to do.
+  }
+  normalizeConf(conf);
+  const { caniuse } = conf;
+  if (!caniuse.feature) {
+    return; // no feature to show
+  }
+  const link = createResourceHint({
+    hint: "preconnect",
+    href: "https://raw.githubusercontent.com",
+  });
+  document.head.appendChild(link);
+  document.head.appendChild(hyperHTML`<style>${caniuseCss}</style>`);
+
+  const headDlElem = document.querySelector(".head dl");
+  const contentPromise = new Promise(async resolve => {
+    let content;
+    try {
+      const stats = await fetchAndCacheJson(caniuse);
+      content = createTableHTML(caniuse, stats);
+    } catch (err) {
+      console.error(err);
+      const msg = `Couldn't find feature "${caniuse.feature}" on caniuse.com? Please check the feature key on [caniuse.com](https://caniuse.com)`;
+      pub("error", msg);
+      content = hyperHTML`<a href="${
+        "https://caniuse.com/#feat=" + caniuse.feature
+      }">caniuse.com</a>`;
+    }
+    resolve(content);
+  });
+  const definitionPair = hyperHTML.bind(document.createDocumentFragment())`
+    <dt class="caniuse-title"
+      id="${`caniuse-${caniuse.feature}`}">Can I Use this API?</dt>
+    <dd class="caniuse-stats">${{
+      any: contentPromise,
+      placeholder: "Fetching data from caniuse.com...",
+    }}</dd>`;
+  headDlElem.appendChild(definitionPair);
```

Comment:
As we are running this module second last, we might as well `await contentPromise` before run() ends. Then we don’t need setTimeout() in the tests, as it’s fairly unreliable.

Reply:
will do that!

Reply:
Not able to run tests for placeholders now though (shown before data being fetched/error). Any tip?

Reply:
Yeah, don't `any: await contentPromise,`, just: 

```
any: contentPromise,
```

And then, underneath:

```
await contentPromise;
```

Reply:
Hopefully should be good now :)

Reply:
The placeholder test still doesn't work :frowning_face: 
Anyway we can throttle the request?

Reply:
heh, crapify :) let me try a few things...

Reply:
Ah, ok... yeah, we won't be able to test for the placeholder because we only get the document after all the plugins have run. That's fine tho. So, let's drop this test.